sdk/python/feast/feature_set.py

```diff
@@ -296,7 +296,7 @@ def drop(self, name: str):
             del self._fields[name]
             return
 
-    def _add_fields(self, fields: List[Field]):
+    def _add_fields(self, fields):
```

Comment:
What happened to the type?

Reply:
There are 2 way of doing this previously:

1. Whenever we create the `_add_fields` the List [feature/entity] need to be converted to field first
2. Remove the type check for the fields part, and let the add handles it. Inside this function, there is a single add function where it checks the type of the field.

Reply:
I am going to increase the scope here a bit, so I apologies for that, but I'd like you to do a quick refactor here.

In a new PR I'd like to make the following changes to remove redundant methods that just add complexity

Can you remove
- `_add_fields()` 
- `add()`

Move the logic from `add` into `entities` and `features` setter.

What do you think? that would make things a lot cleaner, although it will be a breaking change.

Reply:
Ah okay, got your point. Is it, we want to move away from Fields by separating Entities and Features?

Yeah definitely it is way cleaner, but will probably have breaking changes if not careful. 
Will raise the issue to keep track in a new PR

Reply:
Ok, please be descriptive in your issue.